docs/advanced-usage.rst

```diff
@@ -150,32 +148,41 @@ index ``cid``.
 In order to prepare a transfer transaction, Alice needs to provide at least
 three things:
 
-1. ``inputs`` -- one or more conditions that will be fulfilled.
-2. ``asset`` -- the asset being transferred.
-3. ``owners_after`` -- one or more public keys representing the new owner(s).
+1. ``inputs`` -- one or more fulfillments that fulfill a prior transaction's
+   output conditions.
+2. ``asset.id`` -- the id of the asset being transferred.
+3. Recipient ``public_keys`` -- one or more public keys representing the new
+   recipients(s).
 
 To construct the input:
 
 .. ipython::
 
-    In [0]: cid = 0
+    In [0]: fulfilled_output_id = 0
 
-    In [0]: condition = tx['conditions'][cid]
+    In [0]: fulfilled_output = tx['outputs'][fulfilled_output_id]
 
     In [0]: input_ = {
-       ...:     'fulfillment': condition['condition']['details'],
-       ...:     'input': {
-       ...:         'cid': cid,
+       ...:     'fulfillment': fulfilled_output['condition']['details'],
+       ...:     'fulfills': {
+       ...:         'output': fulfilled_output_id,
        ...:         'txid': tx['id'],
        ...:     },
-       ...:     'owners_before': condition['owners_after'],
+       ...:     'owners_before': fulfilled_output['public_keys'],
        ...: }
 
-The asset, can be directly retrieved from the input ``tx``:
+The asset in a ``TRANSFER`` transaction must be a dictionary with an ``id`` key
+denoting the asset to transfer. This asset id can either be the id of the
+``CREATE`` transaction of the asset (as it is in this case), or found inside
+the ``asset`` of a ``TRANSFER`` transaction:
```

Comment:
Perhaps we should make it clear that the asset id is always the id of the transaction in which it was created.